sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3Path.java

```diff
@@ -0,0 +1,569 @@
+/*
```

Comment:
What role does S3Path play in this PR? Beam only requires that you provide a ResourceId, so I suppose S3Path is an internal implementation detail? If so - are all of its functions actually used by the implementation?

Reply:
The PR was originally based on the GCS implementation, which I now see has more complexity (and unused code) than needed. The author probably had something in mind besides just implementing a FileSystem.

I think you'll see that today's changes simplify things. IMHO this should be squashed before merging, and I'm happy to do that whenever.

Reply:
Done